pandas/core/dtypes/cast.py

```diff
@@ -246,6 +246,97 @@ def trans(x):
     return result
 
 
+def maybe_cast_result(
+    result, obj: ABCSeries, numeric_only: bool = False, how: str = ""
```

Comment:
for future reference, typing as ABCSeries is less useful than typing as "Series" via TYPE_CHECKING

Reply:
I attempted making this change, but the mypy is saying that argument 2 of groupby.generic._transform_fast (below) is of type NDFrame. I think that means this should be left as-is, can you confirm?

https://github.com/pandas-dev/pandas/blob/60b0e9fbcfcea7a2bf3e944670ba0c7131c9e332/pandas/core/groupby/generic.py#L522-L532

Reply:
im pretty sure that self.obj is a Series here, so this indicates that we need to get mypy to understand that

Reply:
I added an assert for this and ran the tests; indeed you're right - thanks. I'll be fixing in a subsequent PR.